sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -2731,6 +2731,27 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
       }
     }
   }
+
+  test("spark 50005: Can't overwrite a table that is also being read from") {
+    val tableName1 = "t1"
+    val tableName2 = "t2"
+    withTable(tableName1, tableName2) {
+      sql(s"CREATE TABLE $tableName1 (a STRING, b INT) USING parquet")
+      sql(s"CREATE TABLE $tableName2 (a STRING, b INT) USING parquet")
+      checkError(
+        exception = intercept[AnalysisException] {
+          sql(
+            s"""
+              |INSERT OVERWRITE TABLE $tableName1
+              |SELECT * FROM $tableName2 tb
+              |WHERE NOT EXISTS(SELECT ta.b FROM $tableName1 ta WHERE tb.b = ta.b)
```

Comment:
Do you want to add one more unit test with more nested query to make sure that logic works fine?

Something like:
INSERT OVERWRITE TABLE $tableName1 
  SELECT * FROM $tableName2 tb WHERE NOT EXISTS
    (SELECT * FROM $tableName3 tc WHERE tc.a IN 
      (SELECT td.a FROM $tableName4 td WHERE td.a IN 
        (SELECT ta.a FROM $tableName1 ta)))

Reply:
OK.That should work too.

Reply:
Thank you for this change!